### ISSUE TITLE
Should we retry fetching data on valid 404 results?

### ISSUE THREAD

**Issue by camallen** [zooniverse/panoptes-python-client#196]
I'm looking for a user by id (1613496) but that user deleted their account a while back. However the API responds with a 404. The retry statements shouldn't kick in here and loop asking for the same data as it was a correct response. 

```
Traceback (most recent call last):
  File "testing_api_client.py", line 5, in <module>
    user = User.find(int(1613496))
  File "/Library/Python/2.7/site-packages/panoptes_client/panoptes.py", line 647, in find
    return next(cls.where(id=_id))
  File "/Library/Python/2.7/site-packages/panoptes_client/user.py", line 48, in where
    for user in super(User, cls).where(**kwargs):
  File "/Library/Python/2.7/site-packages/panoptes_client/panoptes.py", line 634, in where
    return cls.paginated_results(*cls.http_get(_id, params=kwargs))
  File "/Library/Python/2.7/site-packages/panoptes_client/panoptes.py", line 585, in http_get
    log_args=False,
  File "/Library/Python/2.7/site-packages/redo/__init__.py", line 165, in retry
    return action(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/panoptes_client/panoptes.py", line 288, in get
    endpoint=endpoint
  File "/Library/Python/2.7/site-packages/panoptes_client/panoptes.py", line 265, in json_request
    json_response['errors']
panoptes_client.panoptes.PanoptesAPIException: Could not find user with id='1613496'
```

**Comment by camallen** [comment]
Just thinknig as valid 404 there is a correct response and can the client raise a new expection type instead of API error - that would allow catching and not retrying this error type vs network / api failure, etc

**Comment by vrooje** [comment]
Would also be useful in messaging to teams who are trying to make author lists for giving volunteer credit and ideally would like to know whether they have an incomplete list and should e.g. try again later.

**Comment by adammcmaster** [comment]
This should have been fixed by 4a68d5aab4d9fc7b7e4286934c2eb8d61dcc83de (when I was refactoring things for `LinkCollection`).
